### ISSUE TITLE
Syntax coloring

### ISSUE THREAD

**Issue by johnnywz00** [toerob/vscode-tads3tools#21]
It's a small matter, but the syntax coloring isn't consistent when within embedded expressions in strings. After a double angle bracket, the coloring may be string coloring, or it may be symbol/statement coloring, and it often switches from one to the other within a single embedded expression.

**Comment by toerob** [comment]
I agree. This will be fixed right after the parser can handle string interpolation, which will fix the nested expressions within strings. It is in good progress and will moat likely be part of the next release.

**Comment by toerob** [comment]
I got stuck with this issue. It was pretty straightforward to add support for string interpolation. Problem arise though when trying to get the small detail of the parser to distinguish between a closing string interpolation within a single quotation and logical shift rights, e.g:

Something like this parses the expression within with no problem: 
`“<a href=“<<var>> >”;`

this on the other side does not:
`“<a href=“<<var>>>”;`
Since the parser gets stuck in the code grammar when it unexpectedly encounters a logical shift right expression “>>>” and can’t  find its way back to the string grammar island again.

Its going to take me a while to figure this out.
My only advice until then is to try to break out interpolates code into functions instead of nestling it.
